build.zig

```diff
@@ -187,9 +177,23 @@ pub fn build(b: *std.build.Builder) void {
     }
 
     {
+        const simulator_options = b.addOptions();
+
+        const StateMachine = enum { testing, accounting };
```

Comment:
I went back and forth on whether I should declare this type somewhere in simulator and _import_ it in build.zig, or rather declare it inline. 

I _think_ that declaring stuff inline is probably the way to go: this defines a cli option of build.zig itself, and also we'll get a separate declaration in the options package anyway. 

To say this more positively, our `const config = @import("./src/config.zig");` in `build.zig` feels a bit hacky to me, as it confusingly triplicates any type `T` declared there: 

* there will be `T` in build.zig
* there will be `T` in tigerbeetle from config
* and there will be similarly-named but different `T` in tigerbeetle from `vsr_options` package

Reply:
:+1: 

Yeah, it's unfortunate that importing in `build.zig` effectively duplicates the type when using options. :disappointed: